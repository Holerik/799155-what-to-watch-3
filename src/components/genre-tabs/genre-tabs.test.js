// footer.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import GenreTabs from './genre-tabs.jsx';

const detailedFilmInfo = [
  {
    id: 0,
    title: `Little Women`,
    poster: `img/little-woman-poster.jpg`,
    altPoster: `ittle Women poster`,
    background: `img/little-wooman-bg.jpg`,
    altBackground: `Movie background`,
    description: `Based on Loiusa May Alcottâ€™s classic novel with the same name, Little Women is a timeless tale of love, life and ambitions`,
    review: `A writer, a sketch artist, an aspiring actress and a budding pianist. All four March sisters are gifted with a unique talent, but they live in the times where the only two choices for women are marriage and death. Writer-director Greta Gerwig`,
    reviews: [1, 2],
    genre: [`Drama`, `Romance`],
    year: 2019,
    duration: `2h 15min`,
    age: `16+`,
    rating: {
      score: `8.0`,
      level: `very good`,
      count: 150,
    },
  }
];
const setPageId = () => {};
const setGenre = () => {};

const tabItems = [`All genres`, `Dramas`, `Crime`];

it(`<GenreTabs /> should render correctly`, () => {
  const tree = renderer
    .create(<GenreTabs
      activeItem={0}
      setPageId={setPageId}
      tabItems={tabItems}
      filmsFullInfo={detailedFilmInfo}
      setGenre={setGenre}
      genre={tabItems[0]}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
