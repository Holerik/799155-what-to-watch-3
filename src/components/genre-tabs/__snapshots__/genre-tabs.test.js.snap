// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreTabs /> should render correctly 1`] = `
<ul
  className="catalog__genres-list"
  onClick={[Function]}
  onMouseOver={[Function]}
>
  <li
    className="catalog__genres-item
                    catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item
                    "
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Dramas
    </a>
  </li>
  <li
    className="catalog__genres-item
                    "
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Crime
    </a>
  </li>
</ul>
`;
